src/__tests__/retry.ts

```diff
@@ -0,0 +1,33 @@
+/**
+ * Stops execution for {@link milliseconds} milliseconds.
+ */
+export async function milliseconds(milliseconds: number): Promise<void> {
+  await new Promise((_) => setTimeout(_, milliseconds));
+}
+
+/**
+ * Executes the function {@link fn}, retrying if it throws an error.
+ * Uses a linear strategy by retrying each {@link delayMs} milliseconds.
+ * If {@link maxAttempts} is reached, it throws the error returned by {@link fn} last execution.
+ */
+export async function retry(
+  fn: () => void,
+  maxAttempts: number = 20,
+  delayMs: number = 200,
+): Promise<void> {
+  let attempt = 0;
+  const execute = async (): Promise<void> => {
+    try {
+      return await fn();
+    } catch (error) {
+      if (attempt >= maxAttempts) {
+        throw error;
+      }
+    }
+
+    await milliseconds(delayMs);
+    attempt++;
+    return execute();
+  };
+  return execute();
+}
```

Comment:
Tests-only utility which takes a function as a parameter and executes it until the execution succeeds. The main point of having something like this is checking state modified by asynchronous processes that are not mocked and are out of Jest scope, like a RabbitMQ queue messages consumption. 

It would be simpler to use a `setTimeout` and check the state was modified after a given amount of time, but setting that time lapse could be problematic and could lead to flaky tests IMO. If the lapse is too short, it might fail sometimes, and if it's too long, it would impact the tests execution time, as we probably need to use this multiple times in the tests.